File: go/abalone/player.go
package abalone

import "encoding/json"

type Player int

const (
	NullPlayer Player = iota
	White
	Black
)

func (p Player) Next() Player {
	switch p {
	case White:
		return Black
	case Black:
		return White
	default:
		return NullPlayer
	}
}

func (p Player) String() string {
	switch p {
	case White:
		return "white"
	case Black:
		return "black"
	default:
		return "null"
	}
}

func (p Player) MarshalJSON() ([]byte, error) {
	return json.Marshal(p.String())
}

func (p *Player) UnmarshalJSON(data []byte) error {
	var s string
	if err := json.Unmarshal(data, &s); err != nil {
		return err
	}
	switch s {
	case "white":
		*p = White
	case "black":
		*p = Black
	default:
		*p = NullPlayer
	}
	return nil
}

File: fe2/abalone-js/src/engine/player.ts
module Abalone {
export module Engine {
	export enum Player {White, Black}
	export function next(p: Player) {
		return (p === Player.White) ? Player.Black : Player.White;
	}

	export function player2str(p: Player): string {
		return p === Player.White ? "white" : "black"
	}
	export function str2player(s: String): Player {
		switch (s) {
			case "white":
			return Player.White;
			case "black": 
			return Player.Black;
			default:
			return null;
		}
	}
}
}